packages/core/src/modules/credentials/services/RevocationService.ts

```diff
@@ -0,0 +1,105 @@
+    this.eventEmitter = eventEmitter
+    this.logger = agentConfig.logger
+  }
+
+  private async processRevocationNotification(
+    revocationRegistryId: string,
+    credentialRevocationId: string,
+    comment?: string
+  ) {
+    const query = { revocationRegistryId, credentialRevocationId }
+    this.logger.trace(`Getting record by query for revocation notification:`, query)
+    const credentialRecord = await this.credentialRepository.getSingleByQuery(query)
+
+    credentialRecord.revocationNotification = new RevocationNotification(comment)
+    await this.credentialRepository.update(credentialRecord)
+
+    this.logger.trace('Emitting RevocationNotificationReceivedEvent')
+    this.eventEmitter.emit<RevocationNotificationReceivedEvent>({
+      type: CredentialEventTypes.RevocationNotificationReceived,
+      payload: {
+        credentialRecord,
+      },
+    })
+  }
+
+  /**
+   * Process a recieved {@link RevocationNotificationMessagev1}. This will create a
+   * {@link RevocationNotification} and store it in the corresponding {@link ConnectionRecord}
+   *
+   * @param messageContext message context of RevocationNotificationMessageV1
+   */
+  public async processRevocationNotificationV1(
+    messageContext: InboundMessageContext<RevocationNotificationMessageV1>
+  ): Promise<void> {
+    this.logger.info('Processing revocation notification v1', { message: messageContext.message })
+    // ThreadID = indy::<revocation_registry_id>::<credential_revocation_id>
+    const threadRegex = /(indy)::(.+)::(\d+)$/
+    const threadId = messageContext.message.issueThread
+    try {
+      const threadIdGroups = threadId.match(threadRegex)
+      if (threadIdGroups) {
+        const [, , revocationRegistryId, credentialRevocationId] = threadIdGroups
+        const comment = messageContext.message.comment
+        await this.processRevocationNotification(revocationRegistryId, credentialRevocationId, comment)
+      } else {
+        throw new AriesFrameworkError(
+          `Incorrect revocation notification threadId format: \n${threadId}\ndoes not match\n"indy::<revocation_registry_id>::<credential_revocation_id>"`
+        )
+      }
+    } catch (error) {
+      this.logger.warn('Failed to process revocation notification message', { error, threadId })
+    }
+  }
+
+  /**
+   * Process a recieved {@link RevocationNotificationMessagev2}. This will create a
+   * {@link RevocationNotification} and store it in the corresponding {@link ConnectionRecord}
+   *
+   * @param messageContext message context of RevocationNotificationMessageV2
+   */
+  public async processRevocationNotificationV2(
+    messageContext: InboundMessageContext<RevocationNotificationMessageV2>
+  ): Promise<void> {
+    this.logger.info('Processing revocation notification v2', { message: messageContext.message })
+    // CredentialId = <revocation_registry_id>::<credential_revocation_id>
+    const credentialIdRegex = /(.*)::(\d+)$/
```

Comment:
I see v1 included an indy: prefix while v2 doesn't. Why is this removed, it would make sense to me to keep it in to be able to detect the type of credential that is revoked?

Reply:
A v1 message looked like
```JSON
{
  "@type": "https://didcomm.org/revocation_notification/1.0/revoke",
  "@id": "<uuid-revocation-notification>",
  "~please_ack": ["RECEIPT","OUTCOME"],
  "thread_id": "<revocation_format>::<revocation_registry_id>::<credential_revocation_id>",
  "comment": "Some comment"
}
```
A v2 message will look like this
```JSON
{
  "@type": "https://didcomm.org/revocation_notification/2.0/revoke",
  "@id": "<uuid-revocation-notification>",
  "~please_ack": ["RECEIPT","OUTCOME"],
  "revocation_format": "<revocation_format>",
  "credential_id": "<revocation_registry_id>::<credential_revocation_id>",
  "comment": "Some comment"
}
```
So we will still be able to detect it from the revocation_format property.

Reply:
The format was moved in v2 from the thread_id field in v2 to a separate field to help with parsing/making it more explicit/clear.

Reply:
I should have read the RFC before making a comment. Thanks for explaining!